### PR TITLE
cross origin timers post

### PR DESCRIPTION
**Pull request by rachelandrew**
Draft of the Aligning timers with cross origin isolation restrictions blog post

### CODE REVIEW

**Comment by rachelandrew** [comment]
@yoavweiss @agektmr here's the PR for you to take a look at.

File: site/en/blog/cross-origin-isolated-hr-timers/index.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+{% endAside %}
+
+Gaining more powerful capabilities by enabling cross-origin isolation is
+[a similar approach posed on `SharedArrayBuffer`](https://developer.chrome.com/blog/enabling-shared-array-buffer/).
+Cross-origin isolation is a state where a web page is isolated from other origins except ones that opted-in.
+
+## Do I need to do something about it?
+
+Probably not. `performance.now()` has been limited to significantly more coarse resolutions in other browsers
+(e.g. 1ms = 0.001 seconds),
```
Comment (agektmr) [inline site/en/blog/cross-origin-isolated-hr-timers/index.md:37]: I'd change `1ms` to `1 millisecond` so the difference with `1 microsecond` is more explicit.

File: site/en/blog/cross-origin-isolated-hr-timers/index.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  - agektmr
+description: Starting in Chrome 91, the resolution of explicit timers will be restricted to 100 microseconds across platforms without cross-origin isolation.
+date: 2021-04-15
+hero: 'image/ZDZVuXt6QqfXtxkpXcPGfnygYjd2/mnoYgEv2E9L6eSThY9G4.jpg'
+alt: A stopwatch
+---
+
+Before version 91,
+[timer resolutions](https://www.w3.org/TR/hr-time/)
+on Chrome were restricted to 5 microseconds on desktop,
```
Comment (yoavweiss) [inline site/en/blog/cross-origin-isolated-hr-timers/index.md:15]: s/on/in/ ?

File: site/en/blog/cross-origin-isolated-hr-timers/index.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+where [site-isolation](https://www.chromium.org/Home/chromium-security/site-isolation) is enabled,
+and to 100 microseconds on Android, where it's not.
+
+Starting from version 91,
+Chrome will be restricting the resolution of explicit [timers](https://www.w3.org/TR/hr-time/)
+(`performance.now()`, `performance.timeOrigin()`,
+and other performance APIs that expose `DOMHighResTimestamps`) to 100 microseconds across platforms.
+By enabling [cross-origin isolation](https://web.dev/coop-coep/),
+websites can relax the restriction to 5 microseconds regardless of platform.
```
Comment (yoavweiss) [inline site/en/blog/cross-origin-isolated-hr-timers/index.md:24]: Just realized that we're not saying that this is following a related spec change: https://github.com/w3c/hr-time/pull/93

Should we?

File: site/en/blog/cross-origin-isolated-hr-timers/index.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+(as well as the availability of SharedArrayBuffers which can be used as implicit timers)
+when [Spectre](https://en.wikipedia.org/wiki/Spectre_(security_vulnerability)) was discovered.
+This is because Spectre,
+and similar speculative execution attacks,
+rely on timers to measure the time certain operations take,
+then guess the contents of the process' memory.
+
+While speculative execution attacks can be executed with coarse timers,
+high-resolution timers can speed them up.
+Chrome introduced
```
Comment (yoavweiss) [inline site/en/blog/cross-origin-isolated-hr-timers/index.md:60]: Site isolation predated Spectre... But the presence of site isolation and other protections (e.g. CORB) enabled Chrome to re-enable these functionalities. Might be worthwhile to make this more precise without diving into the details.

Maybe "Chrome used its site-isolation architecture, as well as other [mechanisms](https://www.chromium.org/Home/chromium-security/corb-for-developers) to mitigate..."

**Review by yoavweiss — APPROVED** [review]
LGTM % comments

File: site/en/blog/cross-origin-isolated-hr-timers/index.md (hunk 3 above)
Comment (rachelandrew) [reply site/en/blog/cross-origin-isolated-hr-timers/index.md:24]: I add this into the latest commit.

**Review by agektmr — APPROVED** [review]
LGTM

File: site/en/blog/cross-origin-isolated-hr-timers/index.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
+---
+layout: 'layouts/blog-post.njk'
+title: Aligning timers with cross origin isolation restrictions
+authors:
+  - yoavweiss
+  - agektmr
+description: Starting in Chrome 91, the resolution of explicit timers will be restricted to 100 microseconds across platforms without cross-origin isolation.
+date: 2021-04-15
```
Comment (agektmr) [inline site/en/blog/cross-origin-isolated-hr-timers/index.md:8]:
```suggestion
date: 2021-04-21
```

**Comment by rachelandrew** [comment]
@mihajlija / @jpmedley this one is to go live today think it just needs an approving review from one of you.

File: site/en/blog/cross-origin-isolated-hr-timers/index.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+Cross-origin isolation provides the standard baseline for browsers to run pages in an isolated environment
+such that they are unable to load unwilling cross-origin resources,
+and therefore, are not at risk for Spectre.
+With cross-origin isolation,
+we can now allow pages to access high-resolution timers,
+`SharedArrayBuffer`,
+and other APIs that are unsafe to expose in processes that can read arbitrary cross-origin data.
+
+To learn more about the background, read
+[Why you need "cross-origin isolated" for powerful features](https://web.dev/why-coop-coep/).
```
Comment (mihajlija) [inline site/en/blog/cross-origin-isolated-hr-timers/index.md:76]:
```suggestion
To learn more about the background behind this change, read
[Why you need "cross-origin isolated" for powerful features](https://web.dev/why-coop-coep/).
```

File: site/en/blog/cross-origin-isolated-hr-timers/index.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+By enabling [cross-origin isolation](https://web.dev/coop-coep/),
+websites can relax the restriction to 5 microseconds regardless of platform.
+
+{% Aside %}
+A microsecond is one millionth of a second (0.000001 seconds). Therefore, 5 microseconds is 0.000005 seconds.
+{% endAside %}
+
+Gaining more powerful capabilities by enabling cross-origin isolation is
+[a similar approach posed on `SharedArrayBuffer`](https://developer.chrome.com/blog/enabling-shared-array-buffer/).
+Cross-origin isolation is a state where a web page is isolated from other origins except ones that opted-in.
```
Comment (mihajlija) [inline site/en/blog/cross-origin-isolated-hr-timers/index.md:32]:
```suggestion
[a similar approach posed on `SharedArrayBuffer`](/blog/enabling-shared-array-buffer/).
Cross-origin isolation is a state where a web page is isolated from other origins except ones that opted-in.
```

File: site/en/blog/cross-origin-isolated-hr-timers/index.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+## Do I need to do something about it?
+
+Probably not. `performance.now()` has been limited to significantly more coarse resolutions in other browsers
+(e.g. 1 millisecond = 0.001 seconds),
```
Comment (mihajlija) [inline site/en/blog/cross-origin-isolated-hr-timers/index.md:37]:
```suggestion
(for example, 1 millisecond = 0.001 seconds),
```

**Review by mihajlija — APPROVED** [review]
LGTM + comments

**Comment by agektmr** [comment]
@rachelandrew can you merge? Happy to do if you need someone.

**Comment by rachelandrew** [comment]
done :)

**Pull request merged**